denote.el

```diff
@@ -4158,7 +4299,10 @@ inserts links with just the identifier."
              (beg (point)))
         (progn
           (denote-link--insert-links files file-type id-only)
-          (denote-link-buttonize-buffer beg (point)))
+          (when (and
+                 denote-fontify-links-mode
+                 (eq denote-link-display-style 'button))
```

Comment:
I think that since we are doing this change, it is an opportunity for us to remove the old functionality. The button was never meant to be there long-term. It was an "okay" solution to have links that are actionable, but using them is not optimal as you already suggest.

Given that `org-fold` is a constraint, maybe we can simplify all this and not offer any option at all? Just go with the `fontify` approach. Then the user can enable the minor mode to activate the links (or we can even make its `:init-value t` if we think that is appropriate).

Reply:
I would also remove the button if backward compatibility is not important to you in this aspect. If you agree with keeping folding, then we can introduce a simple boolean variable to control its behavior. 
Otherwise, the code for `org-fold` can be removed and fontification would be kept. I am happy to make this change, let me know if I should also remove any obsolete code for buttons as well.